### ISSUE TITLE
Cannot connect to any server on macOS BigSure

### ISSUE THREAD

**Issue by phibla** [open62541/open62541#4690]
## Description
I try to run the client_connect example on MacOS BigSure. I would expect that I get the server date but the output look like that:
```
./client_connect opc.tcp://localhost:4840
[2021-10-10 13:36:44.424 (UTC+0200)] warn/userland	AcceptAll Certificate Verification. Any remote certificate will be accepted.
[2021-10-10 13:36:44.426 (UTC+0200)] trace/client	Client connect iterate
[2021-10-10 13:36:49.429 (UTC+0200)] info/client	Client Status: ChannelState: Closed, SessionState: Closed, ConnectStatus: Good
```

In parallel, I checked it on another MacBook with the same operating system. Same result. Afterwards I testet it with Ubuntu 20 and there everything is working.

## Background Information / Reproduction Steps
I followed the official instruction to build the lib:
https://open62541.org/doc/1.2/building.html#building-on-os-x
Afterwards I activate the CMake option UA_BUILD_EXAMPLES with ccmake 
Then I have build the lib again.

Used CMake options:
BUILD_SHARED_LIBS:BOOL=OFF
CLANG_FORMAT_EXE:FILEPATH=CLANG_FORMAT_EXE-NOTFOUND
CMAKE_BUILD_TYPE:STRING=Debug
CMAKE_INSTALL_PREFIX:PATH=/usr/local
CMAKE_OSX_ARCHITECTURES:STRING=
CMAKE_OSX_DEPLOYMENT_TARGET:STRING=
CMAKE_OSX_SYSROOT:PATH=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk
UA_ARCHITECTURE:STRING=posix
UA_BUILD_EXAMPLES:BOOL=ON
UA_BUILD_TOOLS:BOOL=OFF
UA_BUILD_UNIT_TESTS:BOOL=OFF
UA_ENABLE_AMALGAMATION:BOOL=OFF
UA_ENABLE_DA:BOOL=ON
UA_ENABLE_DISCOVERY:BOOL=OFF
UA_ENABLE_DISCOVERY_MULTICAST:BOOL=OFF
UA_ENABLE_ENCRYPTION:STRING=OFF
UA_ENABLE_HISTORIZING:BOOL=OFF
UA_ENABLE_METHODCALLS:BOOL=ON
UA_ENABLE_NODEMANAGEMENT:BOOL=ON
UA_ENABLE_PARSING:BOOL=ON
UA_ENABLE_PUBSUB:BOOL=OFF
UA_ENABLE_PUBSUB_DELTAFRAMES:BOOL=OFF
UA_ENABLE_PUBSUB_ETH_UADP:BOOL=OFF
UA_ENABLE_PUBSUB_INFORMATIONMODEL:BOOL=OFF
UA_ENABLE_PUBSUB_INFORMATIONMODEL_METHODS:BOOL=OFF
UA_ENABLE_SUBSCRIPTIONS:BOOL=ON
UA_ENABLE_SUBSCRIPTIONS_ALARMS_CONDITIONS:BOOL=OFF
UA_ENABLE_SUBSCRIPTIONS_EVENTS:BOOL=OFF
UA_ENABLE_WEBSOCKET_SERVER:BOOL=OFF
UA_FORCE_WERROR:BOOL=ON
UA_LOGLEVEL:STRING=100
UA_MULTITHREADING:STRING=0
UA_NAMESPACE_ZERO:STRING=REDUCED

## Checklist

Please provide the following information:

 - [X] open62541 Version (release number or git tag): ee9d90d7ae3b03478625a6a16443d136bbfe36f8
 - [ ] Other OPC UA SDKs used (client or server):
 - [X] Operating system: macOS BigSure
 - [X] Logs (with `UA_LOGLEVEL` set as low as necessary) attached: UA_LOGLEVEL 100
 - [ ] Wireshark network dump attached
 - [ ] Self-contained code example attached
 - [ ] Critical issue

**Comment by otronarp** [comment]
I'm also seeing this on macOS Big Sur as well as Catalina (on master as well as the latest release v1.2.2). In the server log you can see the that the client opens a connection that is later closed after the default timeout interval of 5 s.
```
[2021-10-19 10:45:14.452 (UTC+0200)] info/network	Connection 6 | New connection over TCP from ::1
[2021-10-19 10:45:19.450 (UTC+0200)] info/network	Connection 6 | Closed
```

The `UA_Client_connect` returns `UA_STATUSCODE_BADTIMEOUT` that I have traced to:
https://github.com/open62541/open62541/blob/ba379f66d98166a19e2374319c739780472f652a/src/client/ua_client_connect.c#L1111

**Comment by wkechel** [comment]
I have this problem too. Are there any solutions available?

**Comment by andreasebner** [comment]
The mentioned code has been revised. If you still have problems, please create a new issue.
